src/js/conntest.js

```diff
@@ -40,32 +40,73 @@ function hostConnectivityTest() {
 }
 
 function runConnectivityTest(iceCandidateFilter, config) {
+  var parsedCandidates = [];
   var call = new Call(config);
   call.setIceCandidateFilter(iceCandidateFilter);
+
+  // Collect all candidate for validation.
+  call.pc1.onicecandidate = function(event) {
```

Comment:
Nit: I prefer addEventListener('icecandidate', {...} ), I find it less error-prone in case there is multiple listeners.
Also it is used way more along this file.